### PR TITLE
v2: Remove deprecated metrics

### LINKED ISSUE #974: v2: Remove deprecated metrics
For the breaking v2 release we want remove all the metrics that have been marked as DEPRECATED.

Blocked until after release v1.9.0.

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #974

### CODE REVIEW

File: docs/cli-arguments.md [hunk 1]
```diff
@@ -25,35 +25,33 @@ spec:
 ```txt
 $ kube-state-metrics -h
 Usage of ./kube-state-metrics:
-      --add_dir_header                              If true, adds the file directory to the header
```
Comment (LiliC) [inline docs/cli-arguments.md]: Besides from here, we also need to remove them from the docs. https://github.com/kubernetes/kube-state-metrics/blob/f09db29f7955e5e628c54130fac1ac116836412c/docs/node-metrics.md

That should fix the CI.

**Comment by yeya24** [comment]
@LiliC  Hi, do I also need to remove this part?
https://github.com/kubernetes/kube-state-metrics/blob/master/docs/README.md#metrics-deprecation

**Comment by LiliC** [comment]
@yeya24 yes, lets remove that as well. 👍 Thanks!

**Comment by LiliC** [comment]
@yeya24 do you mind rebasing your PR, after that we can merge, thanks! :)

**Comment by yeya24** [comment]
@LiliC  HI, could you please take a look at this? CI passed.

**Comment by yeya24** [comment]
@LiliC  Hello,
I have rebased my PR again, is it okay to be merged now?

**Comment by LiliC** [comment]
/hold cancel

Thanks! :)

**Pull request merged**